packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -82,6 +196,88 @@ export const selectTab = (
   };
 };
 
+export const analyzeCollectionSchema = (): CollectionThunkAction<
+  Promise<void>
+> => {
+  return async (dispatch, getState, { dataService, preferences, logger }) => {
+    const { schemaAnalysis, namespace } = getState();
+    const analysisStatus = schemaAnalysis.status;
+    if (analysisStatus === SchemaAnalysisStatus.ANALYZING) {
+      logger.debug(
+        'Schema analysis is already in progress, skipping new analysis.'
+      );
+      return;
+    }
+
+    try {
```

Comment:
I presume it's possible to go to ANALYZING directly from COMPLETED or FAILED as well?

Reply:
That's possible. For now, this function will be invoked when collection metadata is fetched successfully.